src/bots/mascot-bot.js

```diff
@@ -59,6 +80,51 @@ class MascotBot extends SlackBot {
     }
   }
 
+  postMessage(...args) {
+    if (this.debug) {
+      this.log('API limited in debug mode', true);
+    }
+    else {
+      super.postTo(...args);
+    }
+  }
+
+  postTo(name, ...args) {
+    if (this.debug && !DEBUG_WHITELIST.includes(name)) {
+      this.log('postTo API limited in debug mode', true);
+    }
+    else {
+      super.postTo(name, ...args);
+    }
+  }
+
+  postMessageToUser(name, ...args) {
+    if (this.debug && !DEBUG_WHITELIST.includes(name)) {
+      this.log('postMessageToUser API limited in debug mode', true);
+    }
+    else {
+      super.postMessageToUser(name, ...args);
+    }
+  }
+
+  postMessageToGroup(name, ...args) {
+    if (this.debug && !DEBUG_WHITELIST.includes(name)) {
```

Comment:
I feel like there must be some way to DRY this up. Maybe a helper function or something that gets automatically called every time any function is called? idk how this `super` stuff works